sycl/include/syclcompat/math.hpp

```diff
@@ -177,6 +190,36 @@ inline constexpr RetT extend_vbinary2(AT a, BT b, RetT c,
   return sycl::vec<Tint, 2>{temp[0], temp[1]}.template as<sycl::vec<RetT, 1>>();
 }
 
+template <typename RetT, bool NeedSat, bool NeedAdd, typename AT, typename BT,
+          typename BinaryOperation>
+inline constexpr RetT extend_vbinary4(AT a, BT b, RetT c,
+                                      BinaryOperation binary_op) {
+  static_assert(std::is_integral_v<AT> && std::is_integral_v<BT> &&
```

Comment:
I think we can simplify to check whether AT and BT is `int32_t`/`uint32_t`, WDYT?
```
template <class T>
constexpr inline bool is_i32_or_u32 = std::is_same_v<std::decay_t<T>, int32_t> || 
                                      std::is_same_v<std::decay_t<T>, uint32_t>;

```

Reply:
Yes that should look better @yihanwg. We don't want to put it inside the `detail` namespace though, any suggestion?